source/arriving/setup_computer.rst

```diff
@@ -152,12 +152,54 @@ In any case, please refer to the `Github page <https://github.com/scilus/scilpy>
 Containers
 """"""""""
 
-TODO
+Container technologies allow applications and services to be packaged into isolated, self-contained units that can be deployed on any cloud or local host. They are like tiny isolated computers running on other computers. For more information, visit the guide on containers and virtual machines (:ref:`ref_virtualmachines`). A container bundling all the dependencies required for running any flow (see section :ref:`ref_flow`) developped in the lab is available `here <https://hub.docker.com/r/scilus/scilus>`__. In the lab, we use Docker and Apptainer (formerly Singularity) containers, which we describe below.
+
+Docker
+------
+The most popular container technology is undoubtedly Docker. Docker is supported on Mac, Windows and Linux, although it'd seem like Docker running a Linux instance do not work well on MacOS with M1/M2 CPU. The documentation to install Docker is easy enough to follow `(here) <https://docs.docker.com/engine/install/ubuntu>`__.
+
+However, make sure you follow these subsections:
+    - Uninstall old versions (sometimes `this <https://askubuntu.com/questions/935569/how-to-completely-uninstall-docker>`__ is necessary, answer #1)
+    - Install using the apt repository 
+    - Set up the repository
+    - Install Docker Engine
+
+Once installed, you will have to add yourself to the group of users that can run Docker with *sudo* privileges and restart Docker.
+
+.. code-block:: bash
+
+    sudo groupadd docker
+    sudo gpasswd -a ${USER} docker
+    sudo service docker restart
+
+Launch :bash:`docker --version` to verify if it is installed correctly.
+
+Apptainer
+---------
+
+Apptainer (formerly Singularity) is an alternative to Docker. Docker requires root privileges and, as such, is not available on High Performance Computers from the Digital Alliance of Canada (see section :ref:`ref_highperfcomputer`). However, Apptainer containers can run without elevated privileges or root access. Apptainer is particularly useful for saving a Docker container to a file which is then useable in High performance computers, without root privileges.
+
+*Do not try to install on MacOS*
+
+The steps for installing Apptainer on Ubuntu are given below.
```

Comment:
Maybe add "Please refer to the :ref:`ref_highperfcomputer` section if you intend to use Apptainer on High Performance Computers from the Digital Alliance of Canada."

Reply:
Looks to me like we don't say much on containers in the high performance computer section. HPC are already referred on line 180. Sure you want to add this?

Reply:
Maybe my sentence is not the right one, but we don't want people to try and install apptainer on Beluga. I agree that it is not clear in the HPC section, but there is a line for module load apptainer. Maybe just say "Do not try these lines on..."

Reply:
Ok I get it. See my latest commit.